test/unit/stream_stats.test.ts

```diff
@@ -0,0 +1,45 @@
+import { Connection } from "../../src"
+import { expect } from "chai"
+import { Rabbit } from "../support/rabbit"
+import { randomUUID } from "crypto"
+import { expectToThrowAsync, username, password } from "../support/util"
+import { createConnection } from "../support/fake_data"
+
+describe("StreamStats", () => {
+  const rabbit = new Rabbit(username, password)
+  const testStreamName = "test-stream"
+  let connection: Connection
+  let publisherRef: string
+
+  beforeEach(async () => {
+    publisherRef = randomUUID()
+    await rabbit.createStream(testStreamName)
+    connection = await createConnection(username, password)
+  })
+
+  afterEach(async () => {
+    await connection.close()
+    await rabbit.deleteStream(testStreamName)
+  })
+
+  it("gets statistics for a stream", async () => {
+    const publisher = await connection.declarePublisher({ stream: testStreamName, publisherRef })
+    for (let i = 0; i < 5; i++) {
+      await publisher.send(Buffer.from(`test${randomUUID()}`))
+    }
+
+    const stats = await connection.streamStatsRequest(testStreamName)
+
+    expect(stats.committed_chunk_id).to.be.a("BigInt")
+    expect(stats.first_chunk_id).to.be.a("BigInt")
+    expect(stats.last_chunk_id).to.be.a("BigInt")
```

Comment:
Do you think it's possible to make some assertion about the "validity" of data, something like:

I have sent 5 messages committed_chunk_id should be 5?

WDYT?

Reply:
Hi @gpad. I did think about that. In my testing the value of `commited_chunk_id` can vary. For example sending 5 messages, it can be 1, 2, or sometimes 3. 

When I tested with just publishing 1 message in the test, the `commited_chunk_id` varied between -1 and 0.

This behaviour is expected, for example see this doc in the java client https://rabbitmq.github.io/rabbitmq-stream-java-client/snapshot/api/com/rabbitmq/stream/StreamStats.html#committedChunkId()

> The value can be stale as soon as the application reads it though, as the committed chunk ID for a stream that is published to changes all the time.